### PR TITLE
Add support for nameRenderer in RowHeaderCell

### LINKED ISSUE #3362: Add nameRenderer prop to RowHeaderCell
#### Feature request

Hi Blueprint team,

I'm currently working on a Blueprint Table and need to add a Tooltip to the `RowHeaderCell` text. 

There is precedent for this in `ColumnHeaderCell` via the [`nameRenderer`](https://github.com/palantir/blueprint/blob/2b8300b8ce24ad4f17d3e7ef0b3ee0d700e4ae82/packages/table/src/headers/columnHeaderCell.tsx#L31-L44) prop, however this functionality does not exist yet for the `RowHeaderCell`.

If the approach from `ColumnHeaderCell` can be repurposed here, I'm happy to submit a PR. This is a great UI library with a very thoughtful API.

#### Examples

```tsx
<RowHeaderCell
  name="Population"
  nameRenderer={
    (name, index) =>
      <Tooltip content="The population as determined by the 2016 US Census">
        {name}
      </Tooltip>
  } />
```

### PR DESCRIPTION
**Pull request by alecf**
#### Checklist

- [x] Includes tests
- [x] Update documentation

#### Changes proposed in this pull request:

This adds a `nameRenderer` property to `RowHeaderCell` so you can do custom JSX in the row header. This is virtually identical to the `nameRenderer` functionality in the `ColumnHeaderCell`

There were no specific tests for `RowHeaderCell`, so I added them - copying `columnHeaderCell.tsx` directly, and then adapting or removing the tests as appropriate (the row header doesn't have a menu, for example)

#### Reviewers should focus on:

If this will break anything in existing consumers (it shouldn't)

#### Screenshot

No visual impact without the nameRenderer in use, I will try to come up with an example or something

### CODE REVIEW

**Comment by alecf** [comment]
@adidahiya ping?

File: packages/table/src/headers/rowHeaderCell.tsx [hunk 1]
```diff
@@ -22,6 +22,18 @@ export interface IRowHeaderCellProps extends IHeaderCellProps, IProps {
...
     isRowSelected?: boolean;
+
+    /**
+     * A callback to override the default name rendering behavior. The default
+     * behavior is to simply use the `RowHeaderCell`s name prop.
+     *
+     * This render callback can be used, for example, to provide a
+     * `EditableName` component for editing row names.
+     *
+     * The callback will also receive the column index if an `index` was originally
```
Comment (adidahiya) [inline packages/table/src/headers/rowHeaderCell.tsx]: minor typo: this should say "row index"

File: packages/table/test/rowHeaderTests.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the terms of the LICENSE file distributed with this project.
```
Comment (adidahiya) [inline packages/table/test/rowHeaderTests.tsx]: the license has been updated. can you please merge master, delete this comment block, and run the lint auto-fixer to get the new file header?

File: packages/table/src/headers/rowHeaderCell.tsx [hunk 3]
```diff
@@ -30,22 +42,27 @@ export class RowHeaderCell extends AbstractPureComponent<IRowHeaderCellProps, {}
...
             isRowSelected,
+            name,
+            nameRenderer,
 
             // from IHeaderProps
             ...spreadableProps
         } = this.props;
+        const defaultName = <div className={Classes.TABLE_ROW_NAME_TEXT}>{name}</div>;
+
+        const nameComponent = (
```
Comment (adidahiya) [inline packages/table/src/headers/rowHeaderCell.tsx]: nit: I would just inline this on L65 instead of storing in a variable

**Comment by adidahiya** [comment]
@alecf do you still need this?

**Comment by alecf** [comment]
I'll try to return to this - we ended up just not using row headers because of this, but I'd like to return to using them...

**Comment by adidahiya** [comment]
@alec-heif just accept my code suggestions and you're good to go -- you can do it in the UI -- I can't push to your branch

**Pull request merged**